src/main/java/gift/controller/ProductController.java

```diff
@@ -0,0 +1,72 @@
+package gift.controller;
+
+import gift.model.ProductDao;
+
+import java.util.List;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class ProductController {
+
+    private final ProductDao productDao;
+
+    public ProductController(ProductDao productDao) {
+        this.productDao = productDao;
+    }
+
+    @GetMapping("/products/{id}")
+    public ResponseEntity<ProductResponse> getProduct(@PathVariable("id") Long id) {
+        var product = productDao.findById(id)
+            .orElseThrow(() -> new IllegalArgumentException("해당 상품이 존재하지 않습니다."));
+        var response = ProductResponse.from(product);
+        return ResponseEntity.ok(response);
+    }
+
+    @PostMapping("/products")
+    public ResponseEntity<String> createProduct(@RequestBody ProductRequest request) {
+        productDao.insert(request.toEntity());
+        return ResponseEntity.ok().body("Product created successfully.");
+    }
+
+
+    @PutMapping("/products/{id}")
+    public ResponseEntity<String> updateProduct(@PathVariable("id") Long id,
```

Comment:
안고치셔도 되고, 참고차로 공유드립니다! 저는 다음과 같이 변수는 변수대로 개행되는 것을 선호하고, 팀에서도 다음과 같은 방식으로 컨벤션을 맞춥니다ㅎㅎ
코드는 쓰기 작업이 1회라면 읽기 작업이 10회 정도라서, 가독성을 높일 수 있는 개행 및 포매팅을 선호합니다:)

```suggestion
    public ResponseEntity<String> updateProduct(
        @PathVariable("id") Long id,
        @RequestBody ProductRequest request
    ) {
```